### ISSUE TITLE
Jcache: CacheManager#getCache does not register statistics and management bean

### ISSUE THREAD

**Issue by eiden** [ben-manes/caffeine#87]
`CacheManager#createCache` registeres mbeans if it's enabled in the config. But `CacheManager#getCache` does not.

So this test fails:
```
import com.github.benmanes.caffeine.jcache.CacheProxy;
import com.github.benmanes.caffeine.jcache.spi.CaffeineCachingProvider;
import com.typesafe.config.ConfigFactory;
import org.testng.annotations.Test;

import javax.cache.CacheManager;
import javax.cache.Caching;
import javax.cache.spi.CachingProvider;
import javax.management.ObjectName;
import java.lang.management.ManagementFactory;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

public final class TestJmx {

    @Test
    public void testJmxBeanIsRegistered() throws Exception {
        CachingProvider provider = Caching.getCachingProvider(CaffeineCachingProvider.class.getName());
        CacheManager cacheManager = provider.getCacheManager(provider.getDefaultURI(), provider.getDefaultClassLoader());

        CaffeineConfiguration<Object, Object> defaults = TypesafeConfigurator.defaults(ConfigFactory.load());

        CacheProxy<Object, Object> cacheNotInConfigFile =
                (CacheProxy<Object, Object>)cacheManager.createCache("cache-name-not-in-config-file", defaults);

        CacheProxy<Object, Object> cacheInConfigFile =
                (CacheProxy<Object, Object>)cacheManager.getCache("cache-in-config-file");


        assertThat(cacheNotInConfigFile.getConfiguration().isManagementEnabled(), is(true));
        assertThat(cacheNotInConfigFile.getConfiguration().isStatisticsEnabled(), is(true));
        assertThat(cacheNotInConfigFile.getConfiguration().getMaximumSize().getAsLong(), is(10L));

        assertThat(cacheInConfigFile.getConfiguration().isManagementEnabled(), is(true));
        assertThat(cacheInConfigFile.getConfiguration().isStatisticsEnabled(), is(true));
        assertThat(cacheInConfigFile.getConfiguration().getMaximumSize().getAsLong(), is(1L));


        ManagementFactory.getPlatformMBeanServer()
                .getObjectInstance(new ObjectName("javax.cache:Cache=cache-name-not-in-config-file,CacheManager=com.github.benmanes.caffeine.jcache.spi.CaffeineCachingProvider,type=CacheStatistics"));

        //Throws javax.management.InstanceNotFoundException
        ManagementFactory.getPlatformMBeanServer()
                .getObjectInstance(new ObjectName("javax.cache:Cache=cache-in-config-file,CacheManager=com.github.benmanes.caffeine.jcache.spi.CaffeineCachingProvider,type=CacheStatistics"));
    }
}
```

`application.conf`:
```
caffeine.jcache {
  cache-in-config-file {
    policy {
      maximum {
        size: 1
      }
    }
  }

  default {
    policy {
      maximum {
        size: 10
      }
    }
    monitoring {
      statistics = true
      management = true
    }
  }
}
```

**Comment by ben-manes** [comment]
Thanks. It looks like `CacheManagerImpl#getOrCreateCache` should turn it on as well. The change passes the TCK, so I'll port your test over tomorrow and merge the fix in.

**Comment by ben-manes** [comment]
Let me know when you want me to release a new version for you. In the other issue, It sounded like you were going to do a little more testing to try and uncover any other mistakes.

**Comment by eiden** [comment]
It's not super urgent for me to get a new release. I have created a workaround for now. Thanks for the help :)

**Comment by ben-manes** [comment]
Released 2.3.1 with this fix.
